### PR TITLE
Allow passthrough of Redux devtools options

### PR DESCRIPTION
**Pull request by sarimarton**
The current 'devtools()' util in Valtio restricts the Redux-devtools option to the 'name' field, and a custom 'enabled' field. But it might be useful to pass other devtools options too to Redux-devtools. My current case, which motivated this, is that I'm having a large data set in the proxy store (as a ref() of course), and any change to the proxy causes a significant lag in redux-devtools. This has been performance measured in Chrome. The solution is to customize redux-devtools' serializer to skip the large data part, as I don't need to see that part. This way I can further debug other perf-related bottlenecks in my heavy app. During this effort, it seemed like Valtio unnecessarily restricted the redux-devtools options object, so this change only lifts this restriction.

## Related Issues or Discussions

Fixes #

## Check List

- [ ] `yarn run prettier` for formatting code and docs

### CODE REVIEW

File: src/vanilla/utils/devtools.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
+import type { EnhancerOptions } from '@redux-devtools/extension'
```
Comment (dai-shi) [inline src/vanilla/utils/devtools.ts:1]: This produces types in dist. We would need to add it a peer dep.
There is a previous discussion, but the solution is probably copying&pasting types. Or just use `unknown`.

File: src/vanilla/utils/devtools.ts [hunk 2]
```diff
@@ -67,7 +66,7 @@ export function devtools<T extends object>(
   }
 
   let isTimeTraveling = false
-  const devtools = extension.connect({ name })
+  const devtools = extension.connect({ name, ...options })
```
Comment (dai-shi) [inline src/vanilla/utils/devtools.ts:69]: Shouldn't we remove `enabled` from `options`?

File: src/vanilla/utils/devtools.ts (hunk 1 above)
Comment (sarimarton) [reply src/vanilla/utils/devtools.ts:1]: Indeed. I would lean towards peer dep, it can't hurt...

File: src/vanilla/utils/devtools.ts (hunk 2 above)
Comment (sarimarton) [reply src/vanilla/utils/devtools.ts:69]: I agree, I didn't actually figure out exactly why it's there.

Comment (dai-shi) [reply src/vanilla/utils/devtools.ts:69]: `enabled` is used in our lib, right?

**Comment by dai-shi** [comment]
While the idea is good, this PR needs more work and it's incomplete.
Closing as stale for now.
Feel free to add comments.

**Comment by sarimarton** [comment]
Wouldn't it be better merging it in with a @todo flag? I'm not sure what problems need to be solved. Even with no extra typing, just as an escape hatch...

**Comment by dai-shi** [comment]
The two things are really something addressed. Especially, the first one is not acceptable as it caused problems previously. The second one is easy to fix without todo flag.

**Comment by dai-shi** [comment]
I will modify the branch.

File: src/vanilla/utils/devtools.ts [hunk 3]
```diff
@@ -10,10 +10,14 @@ type Message = {
 
 const DEVTOOLS = Symbol()
 
+type EnhancerOptions = Parameters<
+  NonNullable<(typeof window)['__REDUX_DEVTOOLS_EXTENSION__']>['connect']
```
Comment (dai-shi) [inline src/vanilla/utils/devtools.ts:14]: This idea clicked me.

**Review by dai-shi — APPROVED** [review]
Should be good to go.

File: src/vanilla/utils/devtools.ts (hunk 3 above)
Comment (sarimarton) [reply src/vanilla/utils/devtools.ts:14]: Awesome!

**Pull request merged**